plugin/adsk/scripts/mayaUsd_createStagesFromExistingLayer.mel

```diff
@@ -13,17 +13,28 @@
 // limitations under the License.
 //
 
-global string $stageFromExistingLayer_primPath = "";
-global string $stageFromExistingLayer_excludePrimPath = "";
-global int $stageFromExistingLayer_loadPayloads = 1;
+proc setOptionVars (int $forceFactorySettings)
+{
+	if ($forceFactorySettings || !`optionVar -exists stageFromExistingLayer_primPath`) {
+		optionVar -stringValue stageFromExistingLayer_primPath "";
+	}
+
+	if ($forceFactorySettings || !`optionVar -exists stageFromExistingLayer_excludePrimPath`) {
+		optionVar -stringValue stageFromExistingLayer_excludePrimPath "";
+	}
+
+	if ($forceFactorySettings || !`optionVar -exists stageFromExistingLayer_loadPayloads`) {
+		optionVar -intValue stageFromExistingLayer_loadPayloads 1;
+	}        
+}
 
 // stageFromExistingLayer_UISetup
 // creates the options of the stageFromExistingLayer dialog
-global proc stageFromExistingLayer_UISetup(string $parent) {
+global proc string stageFromExistingLayer_UISetup(string $parent) {
     setParent $parent;
-    $parent = `scrollLayout -childResizable true`;
+    string $layout = `scrollLayout -childResizable true`;
 
```

Comment:
Just changing the name of the variable instead of overwriting the one that is passed in because it looks a little cleaner.